src/sdl2_wrapper.cpp

```diff
@@ -652,11 +652,20 @@ CAMLprim value Val_SDL_WindowEventWithData(int type, int windowID, int data1,
   CAMLreturn(ret);
 }
 
+void getNonFocusedMousePosition(SDL_Window *window, int *x, int *y) {
+  int mouseX, mouseY;
+  SDL_GetGlobalMouseState(&mouseX, &mouseY);
+  int windowX, windowY;
+  SDL_GetWindowPosition(window, &windowX, &windowY);
+  *x = mouseX - windowX;
+  *y = mouseY - windowY;
+}
+
 CAMLprim value Val_SDL_Event(SDL_Event *event) {
   CAMLparam0();
   CAMLlocal2(v, vInner);
```

Comment:
Definitely outside the scope of this PR, but these names are terrible. To understand what this new code does I have both scroll up here to the beginning to see their definition, and down to the end of the 200-line long function to see how it's actually used.

Also wondering if there's a reason why the switch cases aren't more self-contained. Since this is C++, not C, you should be able to define new variables and `return` instead of `break`ing, right? Or do the `CAML` macros do weird things that makes this not possible?

Reply:
Definitely agree with you that it's confusing. The `CAML` macros make things harder with regards to returning, because of intricacies with the garbage collector (IIRC). Is there something that you think I should change to make it easier to understand in the mean time? We could write inline functions for each event to make things a little clearer.

Reply:
The [docs](https://caml.inria.fr/pub/docs/manual-ocaml/intfc.html#ss:c-simple-gc-harmony) just say that "Every occurrence of return x must be replaced by CAMLreturn (x)" though, so I would think it should be OK to use in place of the `break`s here.

> Is there something that you think I should change to make it easier to understand in the mean time? We could write inline functions for each event to make things a little clearer.

Nah. I agree it would be cleaner, but too much out of scope for this PR I think. I mention it just as an interesting example to learn from. Perhaps it could be cleaned up later in a separate PR though. I also think giving the variables proper names and using `return` instead of `break` (is possible) would fix most of the problems with this, and is pretty low-hanging fruit.

Reply:
Sounds good - I can rename the variables to something a little more clear.

Reply:
I changed the ambiguous `x, y` in the new commit and changed some other names as well. Let me know what you think!

Reply:
Great! I didn't even notice those were a bit ambiguous, but that definitely helps my understanding!

Reply:
Sounds good. Do you want me to change those `break`s to `return`s in this PR, or do it in a separate one.

Reply:
That and and renaming the `v` and `vInner` variables in a separate PR would be a good idea I think. Assuming it works!

Reply:
Thanks for the help here! Sorry, the unclear variables were my fault. 

I'm not sure if it is possible to use `CAMLreturn` multiple times in a function - I thought it could potentially cause problems, but I guess they do use that in the OCaml runtime, ie: https://github.com/ocaml/ocaml/blob/29d5f485d810c7f88d4513fbffde7a9dcacad3f4/runtime/bigarray.c#L1013 - so it should be OK to refactor this. 

But fine to move to another PR - thank you for the help improving this, @zbaylin !